internal/exec/util/blkid.c

```diff
@@ -40,15 +40,22 @@ static result_t checked_copy(char *dest, const char *src, size_t len) {
 	return RESULT_OK;
 }
 
-result_t blkid_lookup(const char *device, const char *field_name, char *buf, size_t buf_len)
+result_t blkid_lookup(const char *device, bool strict, const char *field_name, char *buf, size_t buf_len)
 {
 	const char *field_val = "\0";
+	int ret;
 
 	blkid_probe pr _cleanup_probe_ = blkid_new_probe_from_filename(device);
 	if (!pr)
 		return RESULT_OPEN_FAILED;
 
-	if (blkid_do_probe(pr) < 0)
+	if (strict)
+		ret = blkid_do_safeprobe(pr);
+	else
+		ret = blkid_do_probe(pr);
+	if (ret == -2)
```

Comment:
This can only be returned by `blkid_do_safeprobe` right?  Better to fold this into the `if (strict)` above?